api/v1alpha1/map_types.go

```diff
@@ -63,6 +63,40 @@ type MapSpec struct {
 	// +kubebuilder:default:=BINARY
 	// +optional
 	InMemoryFormat InMemoryFormatType `json:"inMemoryFormat,omitempty"`
+
+	// EntryListeners contains the configuration for the map-level or entry-based events listeners
+	// provided by the Hazelcast’s eventing framework.
+	// You can learn more at https://docs.hazelcast.com/imdg/latest/events/object-events.
+	// +optional
+	EntryListeners []EntryListenerConfiguration `json:"entryListeners,omitempty"`
+}
+
+type EntryListenerConfiguration struct {
+	// ClassName is the fully qualified name of the class that implements any of the Listener interface.
+	// +kubebuilder:validation:MinLength:=1
+	ClassName string `json:"className"`
+
+	// IncludeValues is an optional attribute that indicates whether the event will contain the map value.
+	// +kubebuilder:default:=true
+	IncludeValues *bool `json:"includeValues"`
```

Comment:
If `IncludeValues` is not optional, we do not need a default value in kubebuilder command. Should we make it optional?

Reply:
To keep it consistent with how we approach other fields, I would suggest we do something like this

```suggestion
        // +kubebuilder:default:=true
	IncludeValues bool `json:"includeValues,omitempty"`
```

I removed the pointer because the field has a default value and added omitempty because default go value and its real default value are not aligned. So, if the resource is created from the operator using this struct, it will have the correct default value.

At the end I would suggest we delete `GetIncludedValue()` function, and just use the field to get its value. WDYT?

Reply:
We are not forcing users to specify the value because we added `+optional` comment along with default kubebuilder tags the fields will be filled in the apiserver side. 

But you are right on the omitempty part. I forgot about user giving `includeValues` as false. In this case the pointer is required for this field. But I still believe the other field does not need to be a pointer.